src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,148 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    static class Node<T> {
+        private T value;
+        private Node<T> prev;
+        private Node<T> next;
+
+        Node(Node<T> prev, T value, Node<T> next) {
+            this.prev = prev;
+            this.value = value;
+            this.next = next;
+        }
```

Comment:
Don't use getters and setters in class Node. Since `Node<T>` is going to be an inner class, you can directly access its private fields from the outer class `MyLinkedList`, making getters and setters redundant.